README.md

```diff
@@ -87,84 +105,104 @@ class ConnectionProvider(Provider):
+If the default scope flow doesn't fit your needs, you can create a custom scope by defining your own `Scope` class.
 
-**Container** is what you use to get your dependency. You just call `.get(SomeType)` and it finds a way to get you an instance of that type. It does not create things itself, but manages their lifecycle and caches. It delegates objects creation to providers which are passed during creation.
+**Container** is what you use to get your dependencies.
+You simply call `.get(SomeType)`, and it finds a way to provide you with an instance of that type.
+The container itself doesn't create objects but manages their lifecycle and caches.
+It delegates object creation to providers added during setup.
 
-**Provider** is a collection of functions which really provide some objects. 
-Provider itself is a class with some attributes and methods. Each of them is either result of `provide`, `alias` or `decorate`. They can be used as provider methods, functions to assign attributes or method decorators.
+**Provider** is a collection of functions that provide concrete objects.
+A `Provider` is a class with attributes and methods, each being the result of `provide`, `alias`, or
+`decorate`.
+They can be used as provider methods, functions to assign attributes, or method decorators.
 
-`@provide` can be used as a decorator for some method. This method will be called when corresponding dependency has to be created. Name of the method is not important: just check that it is different form other `Provider` attributes. Type hints do matter: they show what this method creates and what does it require. All method parameters are treated as other dependencies and created using container.
+`@provide` can be used as a decorator for a method.
+This method will be called when the corresponding dependency has to be created.
+Method name is unimportant: just make sure it's different from other `Provider` attributes.
+Type hints are crucial: they indicate what this method creates and what it requires.
+All method parameters are treated as dependencies and are created using the container.
 
-If `provide` is used with some class then that class itself is treated as a factory (`__init__` is analyzed for parameters). But do not forget to assign that call to some attribute otherwise it will be ignored.
+If `provide` is applied to a class, that class itself is treated as a factory (its `__init__` parameters are analyzed).
+Remember to assign this call to an attribute; otherwise, it will be ignored.
 
-**Component** - is an isolated group of providers within the same container identified by a string. When dependency is requested it is searched only within the same component as its dependant, unless it is declared explicitly.
+**Component** is an isolated group of providers within the same container, identified by a unique string. When a
+dependency is requested, it is only searched for within the same component as the requesting object, unless explicitly
```

Comment:
no, requested object is not clear, it can be treated as the one we used in container `container.get`, here we use direct dependant

Reply:
Thx, fixed